File: AddressBook/react-client/src/dtos/ContactLabel.ts
import Common from "../utility/Common";
import { LabelRes } from "./Label";
import { PagedReq } from "./PagedReq";
import { ContactRes } from "./Contact";

export interface ContactLabelRes {
  contactLabelId?: string;
  labelId?: string;
  label?: LabelRes;
  contactId?: string;
  contact?: ContactRes;
}

export class ContactLabelReqEdit {
  labelId?: string = "";
  contactId?: string = "";
  constructor(contactId?: string) {
    this.contactId = contactId;
  }
}

export class ContactLabelReqSearch extends PagedReq {
  contactId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {contactId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.contactId = contactId;
  }
}

File: AddressBook/Dtos/Dtos/ContactLabel.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class ContactLabelRes
    {
        public int ContactLabelId { get; set; }
        public int? LabelId { get; set; }
        public LabelRes? Label { get; set; }
        public int? ContactId { get; set; }
        public ContactRes? Contact { get; set; }
    }

    public class ContactLabelReqEdit
    {
        public int? ContactId { get; set; }
        public int? LabelId { get; set; }
    }

    public class ContactLabelReqSearch : PagedReq
    {
        [Required]
        public int? ContactId { get; set; }
    }
}
